### ISSUE TITLE
Rich atoms don't cache suspended fields

### ISSUE THREAD

**Issue by JaroslavTulach** [enso-org/enso#8710]
Suspended atom fields, as implemented by
- #6151 

may not be cached properly, when the amount of atom fields is higher than four. Following program shall cache the value generated by the `java.util.Random` generator, but it doesn't:

```ruby
polyglot java import java.util.Random as R

type Num
    Holder a b c d ~num

    new  = Num.Holder "a" "b" "c" "d" (R.new.nextInt)

main =
   fbl = Num.new
   f = fbl.num
   n = fbl.num
   [ f, n, f == n ]
```

It is able to print `[712182466, -522235373, False]`.

**Comment by JaroslavTulach** [comment]
Removing any of the `a`, `b`, `c` or `d` dummy atom fields fixes the problem: `[-1299288440, -1299288440, True]`

**Comment by JaroslavTulach** [comment]
Allocating the `Num.Holder` with primitive types like `Num.Holder 1 2 3 4 (R.new.nextInt)` _doesn't_ fix the problem.
